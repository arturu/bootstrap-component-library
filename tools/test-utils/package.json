{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
