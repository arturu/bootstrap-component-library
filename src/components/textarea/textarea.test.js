import { merge, renderTwigFileAsNode } from "@oe/test-utils";

import demoData from "@oe/data-textarea/data";

const template = "@oe/textarea/textarea.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - textarea", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("renders correctly when required", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, required: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when disabled", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, disabled: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when readonly", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, readonly: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when invalid", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, invalid: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with floating label", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, floating: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly when horizontal", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, horizontal: true })
    ).resolves.toMatchSnapshot();
  });
});
