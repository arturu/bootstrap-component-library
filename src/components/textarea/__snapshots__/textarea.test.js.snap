// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - textarea renders correctly 1`] = `
<label
  class="form-label"
  for="example-textarea-1"
>
  Example textarea
</label>
`;

exports[`OE - textarea renders correctly when disabled 1`] = `
<label
  class="form-label"
  for="example-textarea-1"
>
  Example textarea
</label>
`;

exports[`OE - textarea renders correctly when horizontal 1`] = `
<div
  class="row"
>
  <label
    class="col-form-label col-sm-2"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <div
    class="col-sm-10"
  >
    <textarea
      class="form-control form-control-sm"
      id="example-textarea-1"
      placeholder="Text here"
      rows="4"
    />
  </div>
  <div
    class="form-text"
  >
    An helper text
  </div>
</div>
`;

exports[`OE - textarea renders correctly when invalid 1`] = `
<label
  class="form-label"
  for="example-textarea-1"
>
  Example textarea
</label>
`;

exports[`OE - textarea renders correctly when readonly 1`] = `
<label
  class="form-label"
  for="example-textarea-1"
>
  Example textarea
</label>
`;

exports[`OE - textarea renders correctly when required 1`] = `
<label
  class="form-label"
  for="example-textarea-1"
>
  Example textarea
</label>
`;

exports[`OE - textarea renders correctly with floating label 1`] = `
<div
  class="form-floating"
>
  <textarea
    class="form-control form-control-sm"
    id="example-textarea-1"
    placeholder="Text here"
    rows="4"
  />
  <label
    class="form-label"
    for="example-textarea-1"
  >
    Example textarea
  </label>
  <div
    class="form-text"
  >
    An helper text
  </div>
</div>
`;
